larq_compute_engine/mlir/transforms/optimize.cc

```diff
@@ -58,12 +68,80 @@ DenseElementsAttr Bitpack(PatternRewriter& builder, Attribute x) {
 
 #include "larq_compute_engine/mlir/transforms/generated_optimize.inc"
 
+struct SetBconvReadWriteBitpacked : public OpRewritePattern<TF::LceBconv2dOp> {
+  using OpRewritePattern<TF::LceBconv2dOp>::OpRewritePattern;
+
+  PatternMatchResult matchAndRewrite(TF::LceBconv2dOp bconv_op,
+                                     PatternRewriter& rewriter) const override {
+    Value bconv_input = bconv_op.input();
+    if (!bconv_input.hasOneUse()) return matchFailure();
+
+    auto bconv_input_op =
+        dyn_cast_or_null<TF::LceBconv2dOp>(bconv_input.getDefiningOp());
+    if (!bconv_input_op) return matchFailure();
+
+    const auto inner_tensor_type = bconv_input_op.getType().cast<ShapedType>();
+
+    // We can only apply this transformation if the inner tensor type is F32.
+    if (!inner_tensor_type.getElementType().isF32()) return matchFailure();
```

Comment:
Just to check if I understood it right:
This comment is true because this pass is happening *before* any int8-quantization related passes?

Reply:
Well that's a very good question, to which I don't know the answer. @lgeiger do Int8 passes happen after LCE optimise passes?

Reply:
I guess they *should* happen after, since the int8 stats simply don't make sense when the layers aren't fused. (batchnorm, etc)

Reply:
In any case, I think 47e1084 solves the issue.